chmgen.d

```diff
@@ -566,418 +395,79 @@ main="D Programming Language","d.hhc","d.hhk","chm\index.html","chm\index.html",
-
-<tt>super</tt>
-* [http://www.digitalmars.com/d/2.0/expression.html#PrimaryExpression Expressions]
-* as name of superclass constructor:  [http://www.digitalmars.com/d/2.0/class.html#constructors Classes]
-
-<tt>switch</tt>
-* [http://www.digitalmars.com/d/2.0/statement.html#SwitchStatement Statements]
-
-<tt>synchronized</tt>
-* [http://www.digitalmars.com/d/2.0/statement.html#SynchronizedStatement Statements]
-* as storage class: [http://digitalmars.com/d/2.0/declaration.html Declarations]
-
-
-----
-
-
-<tt>template</tt>
-* [http://www.digitalmars.com/d/2.0/template.html Templates]
-
-<tt>this</tt>
-* [http://www.digitalmars.com/d/2.0/expression.html#PrimaryExpression Expressions]
-* as constructor name:  [http://www.digitalmars.com/d/2.0/class.html#constructors Classes]
-* with ~, as destructor name:  [http://www.digitalmars.com/d/2.0/class.html#destructors Classes]
-
-<tt>throw</tt>
-* [http://www.digitalmars.com/d/2.0/statement.html#ThrowStatement Statements]
-
-<tt>__traits</tt>
-* [http://www.digitalmars.com/d/2.0/traits.html Traits]
-
-<tt>true</tt>
-* [http://www.digitalmars.com/d/2.0/expression.html#PrimaryExpression Expressions]
-
-<tt>try</tt>
-* [http://www.digitalmars.com/d/2.0/statement.html#TryStatement Statements]
-
-<tt>typeid</tt>
-* [http://www.digitalmars.com/d/2.0/expression.html#TypeidExpression Expressions]
-
-<tt>typeof</tt>
-* [http://www.digitalmars.com/d/2.0/declaration.html#typeof Declarations]
-
-
-----
-
-
-<tt>ubyte</tt>
-* [http://www.digitalmars.com/d/2.0/type.html Types]
-
-<tt>ucent</tt>
-* [http://www.digitalmars.com/d/2.0/type.html Types]
-
-<tt>uint</tt>
-* [http://www.digitalmars.com/d/2.0/type.html Types]
-
-<tt>ulong</tt>
-* [http://www.digitalmars.com/d/2.0/type.html Types]
-
-<tt>union</tt>
-* [http://www.digitalmars.com/d/2.0/struct.html Structs & Unions]
-
-<tt>unittest</tt>
-* [http://www.digitalmars.com/d/2.0/unittest.html Unit Tests]
-
-<tt>ushort</tt>
-* [http://www.digitalmars.com/d/2.0/type.html Types]
-
-
-----
+// ************************************************************
 
+void writeTags()
+{
+	stderr.writeln("Writing tags file");
 
-<tt>version</tt>
-* [http://www.digitalmars.com/d/2.0/version.html#version Conditional Compilation]
+	File f;
+	f.open(`d.tag`, "wt");
+	f.writeln("[");
+	foreach (keyword; keywordList)
+	{
+		static struct IndexEntry { string keyword; string[] urls; }
+		IndexEntry entry;
+		entry.keyword = keyword;
+		foreach (url, link; keywords[keyword])
+			if (url in pages)
+				entry.urls ~= `http://dlang.org/` ~ url ~ link.anchor;
+		f.writeln(entry, ",");
+	}
+	f.writeln("]");
+	f.close();
+}
 
-<tt>void</tt>
-* as initializer:  [http://www.digitalmars.com/d/2.0/declaration.html Declarations]
-* as type:  [http://www.digitalmars.com/d/2.0/type.html Types]
+// ********************************************************************
 
+string forwardSlashes(string s)
+{
+	return s.replace(`\`, `/`);
+}
 
-----
+//                   '  *  '
+string backSlashes(string s)
```

Comment:
I went back and forth between slashes and backslashes a few times. I'm not sure which way is more compatible with the CHM readers out there.